lib/screens/app/wallet/receive.dart

```diff
@@ -334,96 +484,286 @@ class _ReceiveFormState extends State<ReceiveForm> {
+            ),
+            cart.length > 0 ? buildCart() : Container(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildCart() {
+    return Padding(
+      padding: const EdgeInsets.only(top: 8.0),
+      child: GridView(
+        physics: ScrollPhysics(),
+        gridDelegate: SliverGridDelegateWithMaxCrossAxisExtent(
+          maxCrossAxisExtent: 200.0,
+          mainAxisSpacing: 10.0,
+          crossAxisSpacing: 10.0,
+        ),
+        shrinkWrap: true,
+        children: [
+          ...cart
+              .map(
+                (product) => GridTile(
+                  header: Container(
+                    padding: EdgeInsets.symmetric(horizontal: 5, vertical: 5),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                      children: [
+                        product.picture.isNotEmpty
+                            ? CircleAvatar(
+                                backgroundImage:
+                                    FileImage(File(product.picture)),
+                                radius: 20,
+                              )
+                            : Container(),
+                        Row(
+                          children: [
+                            Text(
+                              product.price.toString(),
+                              style: TextStyle(
+                                fontSize: 18,
+                                color: Colors.white,
+                                fontWeight: FontWeight.bold,
+                              ),
+                            ),
+                            Image.asset(
+                              'assets/images/seeds.png',
+                              height: 20,
+                            ),
+                          ],
+                        ),
+                      ],
+                    ),
+                  ),
+                  child: Container(
+                    decoration: BoxDecoration(
+                      color: AppColors.getColorByString(product.name),
+                      boxShadow: <BoxShadow>[
+                        BoxShadow(
+                          blurRadius: 15,
+                          color: AppColors.getColorByString(product.name),
+                          offset: Offset(6, 10),
+                        ),
+                      ],
+                    ),
+                    child: Center(
+                      child: Text(
+                        product.name.toString(),
+                        style: TextStyle(
+                          color: Colors.white,
+                          fontSize: 24,
+                          fontWeight: FontWeight.w600,
+                        ),
+                      ),
+                    ),
+                  ),
+                  footer: Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                     children: [
-                      ...cart
```

Comment:
Need to separate out prices for merchant items so they remain the same price

Could now show products in the wrong token

We can store the product with the currency and scoped by currency so just hide

Alternatively, convert the price to another token automatically, but that seems like a lot of work for this experimental feature.